### PR TITLE
Away Mission Madness: Caves 2.0

### PR DESCRIPTION
**Pull request by MMMiracles**
## About The Pull Request
COMPLETELY redoes the Caves away mission from scratch, apart from some vague similarities in base idea, everything else is radically different and upgraded for a modern era.

**List of Content**
- 4 Levels with varying levels of both environmental and mob-based dangers on a (hopefully) scaling basis the further down you go. Jumped in without realizing what hell you just got yourself into? Make it to the end of the 1st floor and take the gateway home, you big baby! The real combat for real men starts on the 2nd floor and lower.
- Multiple segments of the mission use map templates to randomize itself, meaning your first and 5th run on the map won't be entirely the same. Bridges can spawn broken, sections of the map once easy to walk through now has spooky monsters in it! Sure, you can look into the modular map files and look at every possible combination, but you're a giant fucking nerd if you do that!
- Easy to expand on! I had ideas for more than 3 variants per prefab, but I wanted to actually put this out eventually. Extremely easy to create more prefabs that slot into existing locations for additional variety to mission runs.
- 4th floor has an actual boss with an arena picked from multiple prefabs! Watch him monologue about how he'll kill you before he tries to do that! Maybe bring some friends so he doesn't do that!
- Custom content for the away mission itself! Talk to a cool ashwalker who sells you things! Get beat up by cultist mechs! 

## Why It's Good For The Game
We seem to have some initiative to revamp our current pool of away missions to be more interesting and engaging to a larger chunk of the crew without overwhelming them with bodies and/or excessive loot. If we're going to have missions enabled again, those missions should be up to standard with our current mapping guidelines and at least be on-par with the quality of the stations and ruins that already exist. This makes what I feel is a pretty good attempt at creating something that can replayed multiple times while encouraging crew cooperation and less mad-dashing for loot by single individuals.

## Changelog

:cl: MMMiracles
add: The Caves away mission has been fully revamped! Visit your local gateway research initiative for more information. 
/:cl:

**Things That Still Need Done**
- [ ] Finish the boss code, most is done but still need to finish 2nd phase + initial interactions
- [ ] 4th floor work, some work is done but kept back atm since I partially want to keep spoilers while I work on it more
- [ ] Fluff/loot balance pass on all the modules and static map. Every map segment is blocked out and mostly detailed, but still needs a once over for loot and extra details. 
- [ ] More general fluff/lore stuff. I have a chunk done already but I want to do more for fun in-game readings.
- [ ] More trader interactions.
- [ ] Actually do a few balance runs to make sure it isn't too easy/impossible with certain prefab combos.

This has been well over a month in the making, fun stuff all around. I'm pushing a draft now since I'm over the 60-70% completion point and I'd rather get some feedback now on what exists while I work on finishing the rest.

### CODE REVIEW

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm [hunk 1]
```diff
@@ -140,6 +140,8 @@
 			M.projectile_strike(src, target_turf, target, kinetic_gun)
 	if(ismineralturf(target_turf))
 		var/turf/closed/mineral/M = target_turf
+		if(!M.can_mine_with_projectiles)
```
Comment (Fikou) [inline code/modules/projectiles/guns/energy/kinetic_accelerator.dm:143]: we already have strong rocks used by the elephant graveyard no?

**Comment by MMMiracles** [comment]
it shares the same atmos as lavaland so if you're dying on lavaland you're dying in caves

**Comment by ATH1909** [comment]
yooooooooo

File: config/game_options.txt [hunk 2]
```diff
@@ -164,7 +164,7 @@ ALLOW_AI_MULTICAM
 
 ## How long the delay is before the Away Mission gate opens. Default is half an hour.
 ## 600 is one minute.
-GATEWAY_DELAY 18000
+GATEWAY_DELAY 60
```
Comment (san7890) [inline config/game_options.txt:167]: I assume that this is an artefact from local testing- just remove it before you un-draft it (or change the above comment associated to the config flag to match and justify it).

File: code/modules/mob/living/simple_animal/hostile/bosses/clockmaster/phase_one_attacks.dm [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+/datum/action/boss/turret_summon/IsAvailable()
+	. = ..()
+	if(!.)
+		return FALSE
+	return TRUE
+
+/datum/action/boss/turret_summon/Trigger(trigger_flags)
+	if(..())
+		SEND_GLOBAL_SIGNAL(COMSIG_ACTION_TRIGGER_ID,src)
```
[the remark this answers is not in the document]
Comment (MMMiracles) [reply code/modules/mob/living/simple_animal/hostile/bosses/clockmaster/phase_one_attacks.dm:20]: like the arenas i have in mind aren't that big would it be better just to do like a urange or something for this

[the remark this answers is not in the document]
Comment (MMMiracles) [reply code/modules/mob/living/simple_animal/hostile/bosses/clockmaster/phase_one_attacks.dm:20]: i forgot this list exist ngl, will do

**Comment by MMMiracles** [comment]
This is closed until #67953 is resolved, as the mission currently breaks icon smoothing when chosen as a roundstart mission.

**Pull request closed without merging**